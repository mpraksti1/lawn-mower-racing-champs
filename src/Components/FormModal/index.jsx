import React from 'react';
import {
  Route,
  Switch,
  NavLink,
  withRouter,
} from 'react-router-dom';
import PropTypes from 'prop-types';
import LockIcon from 'react-icons/lib/md/lock';
import CreateIcon from 'react-icons/lib/md/create';
import Text from '../../Elements/Text';
import Loader from '../../Elements/Loader';
import LoginForm from '../../Forms/Login';
import RegisterForm from '../../Forms/Register';
import PlayerForm from '../../Forms/Player';
import StyledDiv from './styles';

const FormModal = (props) => {
  const {
    login,
    register,
    createPlayer,
    isRequesting,
    location: { pathname },
  } = props;

  const checkTabStatus = () => pathname !== '/player/new' && !isRequesting;

  return (
    <StyledDiv {...props}>
      {
        checkTabStatus() && (
          <div className="tab-wrapper">
            <NavLink to="/auth/register" className="tab-link area-left">
              <LockIcon />
              <Text {...props} thin sm sans gray>Register</Text>
            </NavLink>
            <NavLink to="/auth/login" className="tab-link area-right">
              <CreateIcon />
              <Text {...props} thin sm sans gray>Login</Text>
            </NavLink>
          </div>
        )
      }
      {
        isRequesting
          ? <Loader />
          : (
            <div className="forms-area">
              <Switch>
                <Route
                  path="/auth/login"
                  render={() => (
                    <LoginForm
                      {...props}
                      login={login}
                    />
                  )}
                />
                <Route
                  path="/auth/register"
                  render={() => (
                    <RegisterForm
                      {...props}
                      register={register}
                    />
                  )}
                />
                <Route
                  path="/player/new"
                  render={() => (
                    <PlayerForm
                      {...props}
                      createPlayer={createPlayer}
                    />
                  )}
                />
              </Switch>
            </div>
          )
      }
    </StyledDiv>
  );
};

FormModal.defaultProps = {
  login: () => {},
  register: () => {},
  createPlayer: () => {},
  pathname: '',
};

FormModal.propTypes = {
  // eslint-disable-next-line
  location: PropTypes.object.isRequired,
  pathname: PropTypes.string,
  isRequesting: PropTypes.bool.isRequired,
  login: PropTypes.func,
  register: PropTypes.func,
  createPlayer: PropTypes.func,
};

export default withRouter(FormModal);
