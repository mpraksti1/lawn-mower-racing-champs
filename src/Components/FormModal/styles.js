import styled, { keyframes } from 'styled-components';

const animateIn = keyframes`
  0% {
    opacity: 0;
    margin-top: -60px;
  }
  100% {
    opacity: 1;
    margin-top: 0;
  }
`;

const StyledDiv = styled.div`
  animation: ${animateIn} 2s ease-out forwards;
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  min-height: 250px;
  width: 350px;
  overflow: hidden;
  transition: all 0.5s ease-in-out;
  background: ${({ theme }) => theme.colors.darkGray};
  margin: 0 auto;
  border-radius: 3px;

  @media (max-width: 767px) {
    background: rgba(51, 51, 51, 0.5);
    max-width: 350px;
    width: 90%;
  }  
  
  .tab-wrapper {
    height: 51px;
    display: grid;
    grid-template-columns: 1fr 1fr;
  }

  .tab-link {
    text-align: center;
    color: #333;
    padding: 16px;
    text-decoration: none;
    background-color: ${({ theme }) => theme.colors.green};
    font-family: ${({ theme }) => theme.fonts.baseFont};
  
    &.active { 
      background-color: transparent;
      color: ${({ theme }) => theme.colors.green};

      span {
        color: ${({ theme }) => theme.colors.green};
      }
    }

    svg {
      margin-right: ${({ theme }) => theme.baseSizeUnit}px;
    }
  }
    
  .forms-area {
    grid-area: main;
    padding: 0 ${({ theme: { baseSizeUnit } }) => (`
      ${baseSizeUnit * 2}px
      ${baseSizeUnit * 2}px 
      ${baseSizeUnit * 2}px
    `)};
    display: grid;
    grid-row-gap: ${({ theme: { baseSizeUnit } }) => baseSizeUnit * 2};
  } 
`;

StyledDiv.displayName = 'FormModalStyled';

export default StyledDiv;
