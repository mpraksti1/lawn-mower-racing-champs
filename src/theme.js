const theme = {
  breakpoints: {
    mobile: 767,
    desktop: 768,
  },
  fonts: {
    baseFont: 'Helvetica',
    sansFont: 'Montserrat',
  },
  baseSizeUnit: 8,
  baseFontUnit: 16,
  colors: {
    yellow: '#f0ff0f',
    red: '#9d0404',
    gray: '#cfcfcf',
    darkGray: '#333',
    green: '#049d63',
    darkGreen: '#097149',
    white: '#eee',
    black: '#111',
    transparentBlack: 'rgba(0, 0, 0, 0.1)',
  },
  fontSizes: {
    xsm: '0.75rem',
    sm: '0.875rem',
    md: '1rem',
    lg: '1.125rem',
    xlg: '1.5rem',
    xxlg: '2.25rem',
  },
};

export default theme;
