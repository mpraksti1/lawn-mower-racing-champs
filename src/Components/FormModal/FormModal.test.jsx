import { MemoryRouter } from 'react-router-dom';
import FormModal from '.';
import theme from '../../theme';

describe('Form modal', () => {
  it('renders without crashing', () => {
    shallow(<FormModal />);
  });

  it('matches snapshot', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/login']} initialIndex={0}>
        <FormModal isRequesting={false} theme={theme} />
      </MemoryRouter>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
