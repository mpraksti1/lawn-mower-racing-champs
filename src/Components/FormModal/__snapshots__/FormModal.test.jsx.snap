// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form modal matches snapshot 1`] = `
<div
  className="sc-ifAKCX VYPsa"
>
  <div
    className="tab-wrapper"
  >
    <a
      aria-current={null}
      className="tab-link area-left"
      href="/auth/register"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m25.2 13.4v-3.4c0-2.8-2.4-5.2-5.2-5.2s-5.2 2.4-5.2 5.2v3.4h10.4z m-5.2 15c1.8 0 3.4-1.6 3.4-3.4s-1.6-3.4-3.4-3.4-3.4 1.6-3.4 3.4 1.6 3.4 3.4 3.4z m10-15c1.8 0 3.4 1.4 3.4 3.2v16.8c0 1.8-1.6 3.2-3.4 3.2h-20c-1.8 0-3.4-1.4-3.4-3.2v-16.8c0-1.8 1.6-3.2 3.4-3.2h1.6v-3.4c0-4.6 3.8-8.4 8.4-8.4s8.4 3.8 8.4 8.4v3.4h1.6z"
          />
        </g>
      </svg>
      <span
        className="sc-bdVaJa gjLHKm"
      >
        Register
      </span>
    </a>
    <a
      aria-current={null}
      className="tab-link area-right"
      href="/auth/login"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"
          />
        </g>
      </svg>
      <span
        className="sc-bdVaJa gjLHKm"
      >
        Login
      </span>
    </a>
  </div>
  <div
    className="forms-area"
  />
</div>
`;
